### PR TITLE
pkg/fqdn: Replace gocheck with built-in go test

### PR DESCRIPTION
**Pull request by sayboras**
Relates: https://github.com/cilium/cilium/issues/28596

### CODE REVIEW

File: pkg/fqdn/dnsproxy/proxy_test.go [hunk 1]
```diff
@@ -269,13 +265,13 @@ func (s *DNSProxyTestSuite) TestRejectFromDifferentEndpoint(c *C) {
...
+	require.NoError(t, err, "Could not update with rules")
 	allowed, err := s.proxy.CheckAllowed(epID2, dstPortProto, dstID1, nil, query)
-	c.Assert(err, Equals, nil, Commentf("Error when checking allowed"))
-	c.Assert(allowed, Equals, false, Commentf("request was not rejected when it should be blocked"))
+	require.NoError(t, err, "Error when checking allowed")
+	require.False(t, allowed, "request was not rejected when it should be blocked")
 }
 
-func (s *DNSProxyTestSuite) TestAcceptFromMatchingEndpoint(c *C) {
+func (s *DNSProxyTestSuite) TestAcceptFromMatchingEndpoint(t *testing.T) {
```
Comment (pippolo84) [inline pkg/fqdn/dnsproxy/proxy_test.go]:
```suggestion
func TestAcceptFromMatchingEndpoint(t *testing.T) {
	s := setupDNSProxyTestSuite(t)

```

**Review by pippolo84 — COMMENTED** [review]
Amazing! :100: 

I've left just one comment inline: it seems a test misses the conversion from `DNSProxyTestSuite` method to an usual unit test.

File: pkg/fqdn/dnsproxy/proxy_test.go (hunk 1 above)
Comment (sayboras) [reply pkg/fqdn/dnsproxy/proxy_test.go]: ah good catch, thanks a lot

**Review by pippolo84 — APPROVED** [review]
:rocket:

**Pull request merged**